android/java/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java

```diff
@@ -207,6 +208,62 @@ public void setGroup(String name, String[] value, EventOptions options, Middlewa
         this.client.setGroup(name, jsonValue, extra);
     }
 
+    public void groupIdentify(String groupType, String groupName, Group event) {
+        this.groupIdentify(groupType, groupName, event, null, null);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, EventOptions options) {
+        this.groupIdentify(groupType, groupName, event, options, null);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, MiddlewareExtra extra) {
+        this.groupIdentify(groupType, groupName, event, null, extra);
+    }
+
+    public void groupIdentify(String groupType, String groupName, Group event, EventOptions options, MiddlewareExtra extra) {
+        if (!this.isInitializedAndEnabled()) {
+            return;
+        }
+        this.handleEventOptions(options, null);
+        com.amplitude.api.Identify identify = new com.amplitude.api.Identify();
+        JSONObject groupProperties = this.getEventPropertiesJson(event);
```

Comment:
1. `this.getEventPropertiesJson` already converts array types to `JSONArray`
2. in the future `this.getEventPropertiesJson` can convert nested object properties to `JSONObject`
3. the code can be removed if `Amplitude-Android` can provide `setGroupProiperties` function similar to `setUserProperties` (not sure how it works with nested objects though)

Reply:
@qingzhuozhen to make the codegen more compact I think it could be nice to add a new `setGroupProperties` to the Amplitude-Android SDK. What do you think? It could be convenient for Amplitude SDK users as well so they can just pass a JSONObject for Group properties rather than having to call `Identify.set` a bunch of times.

Reply:
Yeah, that make sense. Actually other than a new method name, would `groupIdentify(groupType, grouptName, JSONObject groupProperties)` work? I guess this is similar just the name is different. @justin-fiedler

Reply:
@falconandy Can you please add this new method to the core Ampltiude-Android SDK. Thanks

Reply:
@justin-fiedler 
1. It seems I have no write rights to the SDK repo
2. I am not sure what exact environment/IDE should be used for the Amplitude-Android SDK project. I've tried to build/run tests - no luck for now, some warning/error messages.

Reply:
I have set up android sdk env recently. I used the 2021.1.1 version now. Android SDK build tool install version 28.0.3. I had some issues at the beginning but can build the projects now by uninstall and re-install android sdk buildtools. I still have error showing in AmplitudeClient but can build the project.

Reply:
Thank you @bohan-amplitude . Now it works for me. Also I had to change Gradle SDK from default to `openjdk 1.8` to run tests.